From 0874f23b4d09e9e8cafc2893ca312ad300cbbc47 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Fran=C3=A7ois=20Gaffie?= <francois.gaffie@intel.com>
Date: Fri, 20 May 2016 14:34:33 +0200
Subject: [PATCH] tinycompress: compress: Implement avail method
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: François Gaffie <francois.gaffie@intel.com>

diff --git a/include/tinycompress/tinycompress.h b/include/tinycompress/tinycompress.h
index 68626a4..fc02143 100644
--- a/include/tinycompress/tinycompress.h
+++ b/include/tinycompress/tinycompress.h
@@ -108,6 +108,17 @@ struct compress *compress_open(unsigned int card, unsigned int device,
 void compress_close(struct compress *compress);
 
 /*
+ * compress_get_avail: get the hw buffer available
+ * for read/write in bytes.
+ * return 0 on success, negative on error
+ *
+ * @compress: compress stream on which query is made
+ * @avail: buffer availble for write/read, in bytes
+ */
+int compress_get_avail(struct compress *compress,
+        unsigned int *samples);
+
+/*
  * compress_get_hpointer: get the hw timestamp
  * return 0 on success, negative on error
  *
@@ -118,7 +129,6 @@ void compress_close(struct compress *compress);
 int compress_get_hpointer(struct compress *compress,
 		unsigned int *avail, struct timespec *tstamp);
 
-
 /*
  * compress_get_tstamp: get the raw hw timestamp
  * return 0 on success, negative on error
diff --git a/src/lib/compress.c b/src/lib/compress.c
index 84738d2..1db6bde 100644
--- a/src/lib/compress.c
+++ b/src/lib/compress.c
@@ -311,6 +311,20 @@ void compress_close(struct compress *compress)
 	free(compress);
 }
 
+int compress_get_avail(struct compress *compress,
+        unsigned int *avail)
+{
+    struct snd_compr_avail kavail;
+
+    if (!is_compress_ready(compress))
+        return oops(compress, ENODEV, "device not ready");
+
+    if (ioctl(compress->fd, SNDRV_COMPRESS_AVAIL, &kavail))
+        return oops(compress, errno, "cannot get avail");
+    *avail = (unsigned int)kavail.avail;
+    return 0;
+}
+
 int compress_get_hpointer(struct compress *compress,
 		unsigned int *avail, struct timespec *tstamp)
 {
-- 
2.5.0

